dashboard/app/controllers/pd/workshop_enrollment_controller.rb

```diff
@@ -78,6 +78,58 @@ def cancel
     end
   end
 
+  # GET /pd/workshops/join/:section_code
+  def join_section
+    unless current_user
+      redirect_to "/users/sign_in?return_to=#{request.url}"
+      return
+    end
+
+    @workshop = Pd::Workshop.find_by_section_code(params.require(:section_code))
+    unless @workshop
+      render_404
+      return
+    end
+    if workshop_closed?
+      render :closed
+      return
+    elsif workshop_owned_by? current_user
+      render :own
+      return
+    end
+
+    @enrollment = get_workshop_user_enrollment
+  end
+
+  # POST /pd/workshops/join/:section_code
+  # PATCH /pd/workshops/join/:section_code
+  def confirm_join
+    @workshop = Pd::Workshop.find_by_section_code(params.require(:section_code))
+    unless @workshop && current_user
+      render_404
+      return
+    end
+
+    @enrollment = get_workshop_user_enrollment
+    @enrollment.assign_attributes enrollment_params.merge(user_id: current_user.id)
+
+    if @enrollment.valid? && Digest::MD5.hexdigest(@enrollment.email) != current_user.hashed_email
+      @enrollment.errors[:email] = 'must match your login email. If you want to use this email, first go to account settings and update your current email address.'
```

Comment:
@tanyaparker and I talked about that. A link won't show up in the errors the way it's set up now. Also injecting an error here (outside the model) is a little hacky. This might be worth storing in a separate variable and displaying in the view in a friendly manner with a link. I'll take a crack at that on Monday morning, see if I can clean it up fairly quickly.